### ISSUE TITLE
OpenAPIContract.findOperation gives unpredicted results

### ISSUE THREAD

**Issue by szymonsasin-nordic** [eclipse-vertx/vertx-openapi#47]
### Version

4.5.0

### Context

Using vertx-openapi module 
```
implementation("io.vertx:vertx-openapi:4.5.0")
```

### Do you have a reproducer?

No

### Steps to reproduce

Lets assume API with following paths:
_(I know that below API design has an issue, but unfortunately sometimes that is not possible to change.)_
```
GET /users/{id}/addresses/{adr-id}
GET /users/{id}/addresses/foo 
...
```
When using it with:

```kotlin
val contract = OpenApiAuthorizer(jsonContract)
val operation = contract.findOperation("/users/000001/addresses/foo", HttpMethod.GET)
```
Result is not predictable, sometimes it will return wrong `Operation`.

### Extra

Not relevant.

**Comment by pk-work** [comment]
Thanks for finding this. Do you want to provide a fix?

**Comment by CheesyBoy123** [comment]
Hey @pk-work and @szymonsasin-nordic I didn't see any movement on this issue for the last month. I am new to contributing and these seemed "new contributor" friendly. I created this PR https://github.com/eclipse-vertx/vertx-openapi/pull/50 to address this and have the path's resolved more predictably - based on which path matches the input path more exactly (from left to right).

You're reproducer TC is passing now @pk-work I added a few other TCs to make sure everything is covered. Let me know if there is anything that needs to change! This is my first PR for vertx so any comments are welcome 😃
